app/validators/postcode_validator.rb

```diff
@@ -1,11 +1,19 @@
 class PostcodeValidator < ActiveModel::Validator
   VALID_POSTCODE = /^([A-Z]{1,2}[0-9][A-Z0-9]? ?[0-9][A-Z]{2})$/i
-  INVALID_POSTCODE = /^(BF|BX|GIR|XX|GY|JE|IM|AI|GX|KY|VG)/i
+  INVALID_POSTCODE = /^(BF|BX|GIR|XX|GY|JE|IM|AI|GX|VG)/i
+  CAYMANS_POSTCODE = /^KY[0-9]-/i
 
   def validate(record)
-    postcode = PostcodeHelper.normalise(record.postcode)
-    return if postcode.match?(VALID_POSTCODE) && !postcode.match?(INVALID_POSTCODE)
+    return if valid_postcode?(record.postcode) && !invalid_postcode?(record.postcode)
 
     record.errors.add :postcode, "This postcode is invalid"
   end
+
+  def valid_postcode?(postcode)
+    PostcodeHelper.normalise(postcode).match?(VALID_POSTCODE)
+  end
+
+  def invalid_postcode?(postcode)
+    PostcodeHelper.normalise(postcode).match?(INVALID_POSTCODE) || postcode.match?(CAYMANS_POSTCODE)
```

Comment:
Nice work! Could we add Caymans postcode to the invalid regex: `/^(BF|BX|GIR|XX|GY|JE|IM|AI|GX|VG|KY[0-9]-)/i`? Feels a bit cleaner.

Reply:
Ah I see, that's a shame. Thanks for explaining. I've read the commit which explains the reasoning around normalising which seems reasonable. My other thought would be whether we could remove the dash from the normalising list which would mean that the validation code would be cleaner, although may lead to extra db entries / requests to OsPlacesAPI for invalid postcodes containing a dash. On balance, it seems your solution is best 👍